feat(apidoc): split internal/supported API documentation

zregvart opened the pull request:
We need to document what APIs are supported and what APIs are internal.

The easiest way to do that is to do two Swagger executions configured
against supported handlers and internal handlers.

Also switches Maven plugins to a plugin linked from swagger.io.

Now there are two sets of Swagger specifications:
 - internal at /api/v1/internal/swagger.[json|yaml]
 - supported at /api/v1/swagger.[json|yaml]

And two sets of API documents:
 - internal at /api/v1/internal/index.html
 - supported at /api/v1/index.html

To simplify generation IntegrationSupportHandler was moved to it's own
package.

Fixes #1325

chirino commented:
@rhuss: I think we should do that for all syndesis packages that a developer could import into his IDEA.  So basically all the APIs exposed via a tech extension.  Are tech extensions the only users of our java APIs?

zregvart commented:
I've modified the serving of documentation and Swagger files and added an integration test to check.

Let's do `internal` packages in a different PR 🙏 

@chirino @rhuss are we good to merge this in?
